src/main/java/org/gradle/profiler/ProfilerFactory.java

```diff
@@ -37,6 +42,7 @@ public Profiler createFromOptions(OptionSet parsedOptions) {
             put("yourkit-heap", new YourKitHeapAllocationProfilerFactory());
             put("async-profiler", AsyncProfilerFactory.INSTANCE);
             put("async-profiler-heap", AsyncProfilerHeapAllocationProfilerFactory.INSTANCE);
+            put("async-profiler-all", AsyncProfilerAllEventsProfilerFactory.INSTANCE);
```

Comment:
Would it make more sense to have `async-profiler` be the "all" option, and introduce `async-profiler-cpu` as the CPU-only variant?

Reply:
We could potentially have a `-lock` option as well, no? I think it would also make sense to rename `-heap` to `-alloc` to match the naming in Async profiler.

Reply:
Another option would be to ditch the different profilers, and add something like `--async-profiler-events=cpu,lock` with a default to `all`.

Reply:
You already can use `--async-profiler-event` multiple times and that will work. And if you use now `--async-profiler-event alloc` it will use the correct "interval" setting, maybe we should get rid of `async-profiler-heap` and `async-profiler-all` completely.

Reply:
I added a test to show how to specify multiple events from the command line.